### DISCUSSION TITLE
Cannot get twitter user id from jwt.claim

### DISCUSSION THREAD

**Discussion by diasjuniorr** [supabase/supabase#5076] [Questions] [answered]
I'm trying to retrieve the twitter_user_id with the following function but its not working, can anyone help me?

create or replace function auth.twitterid() returns text as $$
  select nullif(current_setting('request.jwt.claim.user_metadata.sub', true), '')::text;
$$ language sql;

**Comment by GaryAustin1** [comment, marked as answer]
You probably need to be using the newer call format in postgrest 9 for your environment.
Edit.  Sorry had posted the wrong link from my clipboard...  here it is: https://postgrest.org/en/stable/api.html#accessing-request-headers-cookies-and-jwt-claims

**Reply by diasjuniorr** [reply]
You're absolutely right! Thank you! Just in case others come looking for the same answer, the correct is:

create or replace function auth.twitterid() returns text as $$
SELECT current_setting('request.jwt.claims', true)::json#>>'{user_metadata,sub}';
$$ language sql;

https://www.postgresql.org/docs/12/functions-json.html

**Reply by kazungusafari** [reply]
Thank you for this.
